src/lib/Shop.js

```diff
@@ -339,74 +360,119 @@ class AutomationShop
      */
     static __internal__buildShopItemList()
     {
+        const sellableItems = {};
+        // Insert this first to maintain the order
         for (const item of pokeMartShop.items)
         {
-            let itemData = { item: item }
+            sellableItems[item.name] = {item: item, towns: [], requirements: []}
+        }
 
-            if ((item.name == "Pokeball") || (item.name == "xAttack") || (item.name == "xClick"))
-            {
-                // Those basic items are always purchasable
-                itemData.isUnlocked = function() { return true; };
-                itemData.isPurchasable = itemData.isUnlocked;
-            }
-            else if (item.name == "Greatball")
+        for (const townName in TownList)
+        {
+            const town = TownList[townName];
+            for (const shop of town.content)
             {
-                itemData.isUnlocked = function() { return TownList["Lavender Town"].isUnlocked(); };
-                itemData.isPurchasable = function()
+                // We only want shops, so discard anything that is not a shop
+                if (shop.items === undefined)
```

Comment:
It would be more readable, and update-friendly to check the class instead
```suggestion
                if (!(shop instanceof Shop))
```

Reply:
I honestly forgot that that was possible. I'm just used to the typeof returning "object" for every kind of object, and forgot that instanceof was a thing.